### PR TITLE
Ft auth service

### PR DESCRIPTION
**Pull request by yetanothersegfault**
This is the implementation of the auth service

### CODE REVIEW

File: fakebook/src/app/services/auth.service.ts [hunk 1]
```diff
@@ -14,17 +14,31 @@ export class AuthService {
   isAuthenticated: boolean | unknown;
 
   constructor(public oktaAuth: OktaAuthService, public router: Router) {
+    this.updateAuthState(false);
+    this.oktaAuth.$authenticationState.subscribe((isAuthenticated) =>
+    this.updateAuthState(isAuthenticated)
```
Comment (escalonn) [inline fakebook/src/app/services/auth.service.ts]: formatting

File: fakebook/src/app/services/auth.service.ts [hunk 2]
```diff
@@ -14,17 +14,31 @@ export class AuthService {
...
+    );
   }
 
   updateAuthState(isAuthenticated: boolean): void {
+    this.isAuthenticated = isAuthenticated;
+    if (isAuthenticated) {
+      this.oktaAuth.getUser().then(console.log);
```
Comment (escalonn) [inline fakebook/src/app/services/auth.service.ts]: why are we doing this? this is just copy pasted code from one of my examples

File: fakebook/src/app/services/auth.service.ts [hunk 3]
```diff
@@ -14,17 +14,31 @@ export class AuthService {
...
+    }
   }
 
   subscribeAuthStateChange(updateFn: (authState: boolean) => void): void {
```
Comment (escalonn) [inline fakebook/src/app/services/auth.service.ts]: this is not how we interact with observables, you just expose the observable object itself for others to subscribe on, we don't need to put a callback layer around the observables.

**Review by RevatureGentry — COMMENTED** [review]
Looks fine, but my concern is by tracking if the user is authenticated in a local variable, when a user refreshes their browser - won't they need to reauthenticate, unless you also use cookies? 
Looks like there's an `isAuthenticated` method on the `OktaAuthService` - setting the local `isAuthenticated` to this result in an `ngOnInit` might better serve you in the future.

File: fakebook/src/app/services/auth.service.ts [hunk 4]
```diff
@@ -14,17 +14,31 @@ export class AuthService {
...
+    this.oktaAuth.$authenticationState.subscribe((authState) => updateFn(authState));
   }
 
   login(): void {
+    this.oktaAuth.signInWithRedirect({
+      originalUri: 'newsfeed'
+    });
```
Comment (escalonn) [inline fakebook/src/app/services/auth.service.ts]: why should this service know or care what route to redirect to? not it's responsibility.

we can compromise process and make non major corrections to the tests in PRs like this too i guess

**Review by escalonn — DISMISSED** [review]

**Review by RevatureGentry — APPROVED** [review]
LGTM

**Pull request merged**